src/olympia/files/tests/test_views.py

```diff
@@ -381,6 +383,14 @@ def test_unicode(self):
         res = self.client.get(self.file_url(u'\u1109\u1161\u11a9'))
         assert res.status_code == 200
 
+    @override_settings(TMP_PATH=unicode(settings.TMP_PATH))
+    def test_unicode_fails_with_wrong_configured_basepath(self):
+        file_viewer = FileViewer(self.file)
+        file_viewer.src = unicode_filenames
+
+        with pytest.raises(UnicodeDecodeError):
+            file_viewer.extract()
+
```

Comment:
https://github.com/mozilla/addons-server/pull/3880/files#diff-9cd11e480f767233e6e5b2c046cddddcR380 is the original test that tests unicode filenames already so both scenarios are covered.